sips/sip-02X/sip-02X-wallet-interface.md

```diff
@@ -0,0 +1,216 @@
+
+# Implementation
+
+## Notes on Serialization
+
+The WebBTC request standard is inspired by JSON-RPC 2.0 for serialization.
+To adhere to a generic serializability, the following notes are given.
+Enums are serialized as humand-readable strings.
+BigInts are serialized as numbers, strings, or anything else that can be parsed by the JavaScript BigInt constructor.
+Bytes are serialized as hex-encoded strings (without the 0x prefix).
+Predefined formats from previous SIPs are used where applicable.
+Addresses are serialized as Stacks c32-encoded strings.
+Clarity values, post-conditions, and transactions are serialized to bytes (defined by SIP-005) and used as hex-encoded strings.
+
+## Methods
+
+This section defines the available methods, their parameters, and result structure.
+Parameters should be considered recommendations for the wallet.
+The user/wallet may choose to ignore/override them.
+Optional params are marked with a `?`.
+
+Methods can be namespaced under `stx_` if used in settings like WebBTC (see WBIP-002).
+In other cases (e.g. WalletConnect), the namespace may already be given by meta-data (e.g. a `chainId` field) and can be omitted.
+
+### Method Independent
+
+#### Transfer & Transaction Definitions
+
+`params`
+
+- `network?`: `'mainnet' | 'testnet' | 'regtest' | 'mocknet'`
+- `fee?`: `number | string` (anything parseable by the BigInt constructor)
+- `nonce?`: `number | string` (anything parseable by the BigInt constructor)
+- `attachment?`: `string` hex-encoded
+- `anchoreMode?`: `'onChainOnly' | 'offChainOnly' | 'any'`
+- `postConditionMode?`: `'allow' | 'deny'`
+- `postConditions?`: `PostCondition[]`, defaults to `[]`
+- `sponsored?`: `boolean`, defaults to `false`
+- `sender?`: `string` address, Stacks c32-encoded
+- ~~`appDetails`~~ _removed_
+- ~~`onFinish`~~ _removed_
+- ~~`onCancel`~~ _removed_
+
+`where`
+
+- `PostCondition`: `string` hex-encoded
+
+### `transferStx`
+
+`params`
+
+- `recipient`: `string` address, Stacks c32-encoded
+- `amount`: `number | string` (anything parseable by the BigInt constructor)
+- `memo?`: `string`, defaults to `''`
+
+`result`
+
+- `txId`: `string` hex-encoded
+- `txRaw`: `string` hex-encoded
+
+### `transferFt`
+
+`todo: haven't existed yet, should we add them?`
+
+### `transferNft`
+
+`todo: haven't existed yet, should we add them?`
+
+### `contractCall`
+
+`params`
+
+- `contractAddress`: `string` address, Stacks c32-encoded
+- `contractName`: `string`
+- `functionName`: `string`
+- `functionArgs`: `ClarityValue[]`, defaults to `[]`
+
+`where`
+
+- `ClarityValue`: `string` hex-encoded
+
+`result`
+
+- `txId`: `string` hex-encoded
+- `txRaw`: `string` hex-encoded
+
+### `contractDeploy`
+
+`params`
+
+- `contractName`: `string`
+- `codeBody`: `string` Clarity contract code
+- `clarityVersion?`: `number`
+
+`result`
+
+- `txId`: `string` hex-encoded
```

Comment:
IMO in Bitcoin the more common treatment of this is all lower `txid`, as it's treated as a single concept or word https://github.com/search?q=repo%3Abitcoin%2Fbitcoin%20txid&type=code

Reply:
I agree. `txid` is nice, should we also do `txRaw`/`txHex` → `tx` (or `transaction`)?

Reply:
I think that makes makes yeah. Typings make describing the format largely unnecessary.